app/src/main/java/fr/free/nrw/commons/media/ZoomableActivity.kt

```diff
@@ -163,13 +165,17 @@ class ZoomableActivity : BaseActivity() {
             handleResult(it)
         }
 
-        if(prefs.getBoolean(CustomSelectorConstants.FULL_SCREEN_MODE_FIRST_LUNCH, true)) {
-            // show welcome dialog on first launch
-            showWelcomeDialog()
-            prefs.edit().putBoolean(
-                CustomSelectorConstants.FULL_SCREEN_MODE_FIRST_LUNCH,
-                false
-            ).apply()
+        val origin = intent.getStringExtra(originLabel);
+
+        if (origin == null) {
```

Comment:
Please add a comment explaining each of the two cases that can happen. Thanks!

Reply:
Done

Reply:
For extra reader-friendliness, could you please rephrase to `If origin is "null" it means that ZoomableActivity was created by the custom picker (rather than by MediaDetailsFragment) so we need to show the first time popup in full screen mode if needed.`

Thanks!